src/rpc/multipass.proto

```diff
@@ -204,26 +208,66 @@ message InstanceStatus {
     Status status = 1;
 }
 
+message InstanceSpecificDetails {
+    string image_release = 1;
+    string current_release = 2;
+    string id = 3;
+    string load = 4;
+    string memory_usage = 5;
+    string disk_usage = 6;
+    repeated string ipv4 = 7;
+    repeated string ipv6 = 8;
+    int32 num_snapshots = 9;
+}
+
+message SnapshotFundamentals {
+    string snapshot_name = 1;
+    string parent = 2;
+    string comment = 3;
+    google.protobuf.Timestamp creation_timestamp = 4;
+}
+
+message SnapshotSpecificDetails {
+    SnapshotFundamentals fundamentals = 1;
+    string size = 2;
+    repeated string children = 4;
+}
+
+message DetailedInfoItem {
+    string instance_name = 1;
+    InstanceStatus instance_status = 2;
+    string memory_total = 3;
+    string disk_total = 4;
+    string cpu_count = 5;
+    MountInfo mount_info = 6;
+
+    oneof extra_info {
+        InstanceSpecificDetails instance_info = 7;
+        SnapshotSpecificDetails snapshot_info = 8;
+    }
+}
+
+message SnapshotOverviewInfoItem {
+    string instance_name = 1;
+    SnapshotFundamentals fundamentals = 2;
+}
+
+message DetailedInfoReport {
+    repeated DetailedInfoItem details = 1;
+}
+
+message SnapshotOverviewInfoReport {
+    repeated SnapshotOverviewInfoItem overview = 1;
+}
+
 message InfoReply {
-    message Info {
-        string name = 1;
-        InstanceStatus instance_status = 2;
-        string image_release = 3;
-        string current_release = 4;
-        string id = 5;
-        string load = 6;
-        string memory_usage = 7;
-        string memory_total = 8;
-        string disk_usage = 9;
-        string disk_total = 10;
-        repeated string ipv4 = 11;
-        repeated string ipv6 = 12;
-        MountInfo mount_info = 13;
-        string cpu_count = 14;
-        int32 num_snapshots = 15;
+    oneof info_contents
+    {
+        DetailedInfoReport details = 1;
```

Comment:
Hmm, good point. However, overview is info too. How about `detailed_report`? It is a little longer, but contrasts with `snapshot_overview` and I guess `reply.detailed_report().details()` is OK.

Reply:
Sure, I'm ok with that. Can always be changed at a later date. Right now, it would just be nice to get this merged into the feature branch so that other items can be built off of it.